api/utxoCache.ts

```diff
@@ -0,0 +1,134 @@
+import { UtxoOrdinalBundle } from '../types/api/xverse/ordinals';
+import { getAddressUtxoOrdinalBundles, getUtxoOrdinalBundle } from './ordinals';
+
+export type UtxoCacheStruct = {
+  [utxoId: string]: UtxoOrdinalBundle;
+};
+
+type UtxoCacheStorage = {
+  version: number;
+  utxos: UtxoCacheStruct;
+};
+
+export type StorageAdapter = {
+  get(key: string): Promise<string | null> | string | null;
+  set(key: string, value: string): Promise<void> | void;
+  remove(key: string): Promise<void> | void;
+};
+
+type UtxoCacheConfig = {
+  cacheStorageController: StorageAdapter;
+};
+
+export class UtxoCache {
+  private readonly _cacheStorageController: StorageAdapter;
+
+  utxos: UtxoCacheStruct = {};
+
+  VERSION = 1;
+
+  constructor(config: UtxoCacheConfig) {
+    this._cacheStorageController = config.cacheStorageController;
+  }
+
+  private getAddressCacheStorageKey = (address: string) => `utxoCache-${address}`;
+
+  _getCache = async (address: string): Promise<UtxoCacheStorage> => {
+    const cache = await this._cacheStorageController.get(this.getAddressCacheStorageKey(address));
+    if (cache) {
+      return JSON.parse(cache);
+    } else {
+      return {
+        version: -1,
+        utxos: {},
+      };
+    }
+  };
+
+  private _setCache = async (utxos: UtxoCacheStruct, address: string): Promise<void> => {
+    await this._cacheStorageController.set(
+      this.getAddressCacheStorageKey(address),
+      JSON.stringify({
+        version: this.VERSION,
+        utxos: utxos,
+      }),
+    );
+  };
+
+  _getAddressUtxos = async (address: string) => getAddressUtxoOrdinalBundles(address, 0, 60);
```

Comment:
```suggestion
  private _getAddressUtxos = async (address: string) => getAddressUtxoOrdinalBundles(address, 0, 60);
```

Reply:
Maybe we should initiate the cache with all the UTXOs in the address instead of just the first 60